servicetalk-client-api/src/main/java/io/servicetalk/client/api/LoadBalancerFactory.java

```diff
@@ -35,8 +35,36 @@
      * and hence will call {@link ConnectionFactory#closeAsync()} when {@link LoadBalancer#closeAsync()} is called.
      * @param <T> Type of connections created by the passed {@link ConnectionFactory}.
      * @return a new {@link LoadBalancer}.
+     * @deprecated In the future only {@link #newLoadBalancer(String, Publisher, ConnectionFactory)} will remain,
+     * please use that method instead.
      */
+    @Deprecated
     <T extends C> LoadBalancer<T> newLoadBalancer(
             Publisher<? extends ServiceDiscovererEvent<ResolvedAddress>> eventPublisher,
             ConnectionFactory<ResolvedAddress, T> connectionFactory);
+
+    /**
+     * Create a new {@link LoadBalancer}.
+     * <p>
+     * Note this method has a default implementation to not break the {@link FunctionalInterface} contract, however
+     * in a future release the other deprecated {@link #newLoadBalancer(Publisher, ConnectionFactory) method}
+     * will be removed and this method will become the functional interface.
+     * @param targetResource A {@link String} representation of the target resource for which the created instance
+     * will perform load balancing. Bear in mind load balancing is performed over the a collection of hosts provided
+     * via the {@code eventPublisher} which may not correspond directly to a single unresolved address, but potentially
+     * a merged collection.
+     * @param eventPublisher A stream of {@link ServiceDiscovererEvent}s which the {@link LoadBalancer} can use to
+     * connect to physical hosts. Typically generated from a {@link ServiceDiscoverer}.
+     * @param connectionFactory {@link ConnectionFactory} that the returned {@link LoadBalancer} will use to generate
+     * new connections. Returned {@link LoadBalancer} will own the responsibility for this {@link ConnectionFactory}
+     * and hence will call {@link ConnectionFactory#closeAsync()} when {@link LoadBalancer#closeAsync()} is called.
+     * @param <T> Type of connections created by the passed {@link ConnectionFactory}.
+     * @return a new {@link LoadBalancer}.
+     */
+    default <T extends C> LoadBalancer<T> newLoadBalancer(
+            String targetResource,
+            Publisher<? extends ServiceDiscovererEvent<ResolvedAddress>> eventPublisher,
```

Comment:
We have another task on the short roadmap to change this input parameter from `Publisher<SDE>` to `Publisher<Collection<SDE>>` to make it consistent with `SD#discover(...)` API. It will force us to deprecate this method one more time. To minimize deprecations in 0.41 we should not cherry-pick this change into 0.41 until both PRs are ready.